Ikuyadeu/vscode-R
# Inject R Markdown features into Markdown grammar

Closes #220 (HTML comments)

Closes #116 (linebreaks)

Closes #48 (block content in lists)

Closes #36 (citations, list items, YAML frontmatter)

**What problem did you solve?**

VSCode provides a Markdown grammar, but it was being overruled by our R Markdown grammar. This PR uses the Markdown grammar, and injects just the necessary R Markdown features (chunks with `{r}` etc., inline R code).

**Screenshots**

Before:

![example1_before](https://user-images.githubusercontent.com/23294156/74223727-af9c1c00-4cfa-11ea-848c-5bfd723e16fe.jpg)

After:

![example1_after](https://user-images.githubusercontent.com/23294156/74223788-cd698100-4cfa-11ea-8d6a-dec8b8cb5196.JPG)

**How can I test this pull request?**

1. Create an Rmd file with this text (removing leading spaces if necessary):

```
    ---
    title: YAML header
    ---

    # Inline R

    `r x <- 1`

    # R Markdown R block

    ```{r}
    head(iris)
    ```

    # C++ block

    ```{cpp}
    int x = 0;
    ```
```

2. Place cursor on line 1. Use `Edit -> Toggle Line Comment`. Verify that comment is HTML comment ``
2. Place cursor in R block. Use `Edit -> Toggle Line Comment`. Verify that comment is R comment `#`
2. Place cursor in C++ block. Use `Edit -> Toggle Line Comment`. Verify that comment is C++ comment `//`

Previously, all these comments were R style `#`.
